src/front/glsl/lex_pp.rs

```diff
@@ -0,0 +1,242 @@
+                    "for" => Token::For(meta),
+                    // types
+                    "void" => Token::Void(meta),
+                    "const" => Token::Const(meta),
+
+                    word => match parse_type(word) {
+                        Some(t) => Token::TypeName((meta, t)),
+                        None => Token::Identifier((meta, String::from(word))),
+                    },
+                }
+            }
+            //TODO: unsigned etc
+            TokenValue::Integer(integer) => Token::IntConstant((meta, integer.value as i64)),
+            TokenValue::Punct(punct) => match punct {
+                // Compound assignments
+                Punct::AddAssign => Token::AddAssign(meta),
+                Punct::SubAssign => Token::SubAssign(meta),
+                Punct::MulAssign => Token::MulAssign(meta),
+                Punct::DivAssign => Token::DivAssign(meta),
+                Punct::ModAssign => Token::ModAssign(meta),
+                Punct::LeftShiftAssign => Token::LeftAssign(meta),
+                Punct::RightShiftAssign => Token::RightAssign(meta),
+                Punct::AndAssign => Token::AndAssign(meta),
+                Punct::XorAssign => Token::XorAssign(meta),
+                Punct::OrAssign => Token::OrAssign(meta),
+
+                // Two character punctuation
+                Punct::Increment => Token::IncOp(meta),
+                Punct::Decrement => Token::DecOp(meta),
+                Punct::LogicalAnd => Token::AndOp(meta),
+                Punct::LogicalOr => Token::OrOp(meta),
+                Punct::LogicalXor => Token::XorOp(meta),
+                Punct::LessEqual => Token::LeOp(meta),
+                Punct::GreaterEqual => Token::GeOp(meta),
+                Punct::EqualEqual => Token::EqOp(meta),
+                Punct::NotEqual => Token::NeOp(meta),
+                Punct::LeftShift => Token::LeftOp(meta),
+                Punct::RightShift => Token::RightOp(meta),
+
+                // Parenthesis or similar
+                Punct::LeftBrace => Token::LeftBrace(meta),
+                Punct::RightBrace => Token::RightBrace(meta),
+                Punct::LeftParen => Token::LeftParen(meta),
+                Punct::RightParen => Token::RightParen(meta),
+                Punct::LeftBracket => Token::LeftBracket(meta),
+                Punct::RightBracket => Token::RightBracket(meta),
+
+                // Other one character punctuation
+                Punct::LeftAngle => Token::LeftAngle(meta),
+                Punct::RightAngle => Token::RightAngle(meta),
+                Punct::Semicolon => Token::Semicolon(meta),
+                Punct::Comma => Token::Comma(meta),
+                Punct::Colon => Token::Colon(meta),
+                Punct::Dot => Token::Dot(meta),
+                Punct::Equal => Token::Equal(meta),
+                Punct::Bang => Token::Bang(meta),
+                Punct::Minus => Token::Dash(meta),
+                Punct::Tilde => Token::Tilde(meta),
```

Comment:
it seems rather awkward that most (if not all) token variants come with `meta`. Can we instead just place them side by side? I.e. `(Token, Meta)` or a structure with fields, etc

Reply:
this is (unfortunately) entirely controlled by pomelo

Reply:
Resolving for now - until we remove pomelo :)